pandas/tests/extension/base/setitem.py

```diff
@@ -357,7 +357,7 @@ def test_setitem_with_expansion_dataframe_column(self, data, full_indexer):
         key = full_indexer(df)
         result.loc[key, "data"] = df["data"]
 
-        tm.assert_frame_equal(result, expected)
+        tm.assert_frame_equal(result, expected, check_column_type=False)
```

Comment:
IMO might as well rename `"data"` to 0 to avoid this

Reply:
sure